crates/astria-composer/src/searcher/collector.rs

```diff
@@ -0,0 +1,221 @@
+use std::time::Duration;
+
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use ethers::providers::{
+    Provider,
+    ProviderError,
+    Ws,
+};
+use humantime::format_duration;
+use tokio::sync::{
+    mpsc::{
+        error::SendTimeoutError,
+        Sender,
+    },
+    watch,
+};
+use tracing::{
+    debug,
+    info,
+    instrument,
+    warn,
+};
+
+/// A wrapper around an [`ethers::types::Transaction`] that includes the chain ID.
+///
+/// Used to send new transactions to the searcher.
+pub(super) struct Transaction {
+    pub(super) chain_id: String,
+    pub(super) inner: ethers::types::Transaction,
```

Comment:
this is eth-specific, do we plan to make this generic later?

Reply:
Yeah, I think that's the plan ultimately. But not here. :)